rfcs/text/0014_notification_service.md

```diff
@@ -0,0 +1,401 @@
+look up through the storage to remove notifications with *expire_at* > the current timestamp.
+
+Data are stored in the hidden system index to prevent accidental access by a random user.
+We don’t expect notifications to contain sensitive information.
+They must be stored in an encrypted form otherwise. 
+Kibana can query user-specific notifications by *recipent_id*.
+
+Alternative implementation might be to use SO with OLS([#17888](https://github.com/elastic/kibana/issues/17888)) and built-in migration support. The main obstacle to use SO is limited search functionality that doesn’t allow using complex search queries (can be addressed in [#84729](https://github.com/elastic/kibana/issues/84729)).
+
+### RemoteRepository
+Cloud plugin registered RemoteRepository as a strategy Kibana Notification Service for creation, storage,
+and retrieval notifications from the remote Unified Notification Service. 
+It’s still not clear what functionality is supported by UNS, but we expect it to be:
+- Create a notification for a particular recipient, a group of recipients. 
+- Read a list notification for a particular recipient.
+- Filter notification list.
+- Change a notification state.
+
+Kibana-specific entities shouldn’t leak to UNS: 
+- Kibana sends message translations for all the supported locales.
+- Kibana-specific groups of recipients unfolded to a list of *recipient_id*.
+- *recipient_id* is set to *elastic_id*.
+
+## Delivery
+The delivery mechanism is responsible for the notifications shipment to Kibana UI. 
+
+#### List of notifications
+For simplicity, we can start with Kibana UI using the HTTP polling mechanism to fetch the last notifications from `/api/notifications` endpoint.
+Server orders notifications by *pinned* and *created_at* timestamp.
+
+Kibana HTTP API:
+- Endpoint: `GET /api/notifications/` 
+    - `search_after?: number` - Unix timestamp to start searching from.
+    - `type_id?: string[]` - list of source types to filter by.
+    - `size?: number` - max number of notifications to retrieve.
+
+*search_after* and *size* params used to implement pagination logic.
+
+In one of the next phases of the project, we will have to do load testing and decide whether to consider the alternative
+to HTTP polling delivery mechanisms: passing information with a response header (as done for licensing), WebSockets, ServerSideEvents.
+
+#### Notification Status
+The delivery mechanism handles Notification status change performed by Kibana UI.
+Every notification has a different status for every user.
+Supported notification statuses:
+- `is_read: boolean`
+- `is_pinned: boolean`
+
+Kibana UI changes notification status to `is_read: true` when a user interacts with it: clicks on CTA or `Mark as Read` button.
+
+The user can change the notification status to `is_read: false` manually in the Kibana UI.
+
+The user can change the status for all the existing notifications (`mark all as read`), even for those not-rendered in UI yet.
+
+Kibana HTTP API:
```

Comment:
I'm assuming these HTTP APIs will be implicitly scoped to the currently authenticated recipient because none of them have the `recipient_id` as a parameter?

Reply:
Yes. Otherwise, someone might have access to another person's notifications.

Reply:
I think it's fine. I was just trying to think through how this would work with Cloud's UNS, and whether or not there would be any situations where Cloud would be accessing Kibana's API and wanting to specify an explicit user id, but I haven't thought of any as all Cloud interactions should go through the Cloud UNS APIs.